Objects/longobject.c

```diff
@@ -769,6 +769,36 @@ PyLong_AsUnsignedLongMask(PyObject *op)
     return val;
 }
 
+int
+PyLong_IsPositive(PyObject *obj)
+{
+    if (!PyLong_Check(obj)) {
+        PyErr_Format(PyExc_TypeError, "expect int, got %T", obj);
```

Comment:
I do see that this error message was copied, but I *really* don't like it. I would rather use the repr instead of the type.

Reply:
@rruuaanng, please address all of Peter's remarks.

Reply:
@ZeroIntensity, what do you mean?

If you suggest to report the repr of the instance, this is a large no. The check is type depending, not value depending, so we should use TypeError and the type name. Also, the repr of the instance can be arbitrary long and execute arbitrary code -- both are undesirable.

If you suggest to report the repr of the type object, the type name is nicer looking and is enough in most cases. It is also faster and has less side effects.

Reply:
There seems to be a little misunderstanding. I answered all the reviews below.

Reply:
Yeah, I see those benefits. The main downside to using the type name is that it's painful when debugging, because you don't exactly know *what* object it was that you passed. It's quite rare for the repr to have side effects. I guess I'm happy to go for the type name (though, we use a different format for it normally--I would rather it mimic `PyArg_ParseTuple`).

Reply:
The value of the object is irrelevant for TypeError. Only its type matters. If this is not so, TypeError is a wrong exception type.